src/careamics/models/lvae/noise_models.py

```diff
@@ -1,71 +1,52 @@
 import json
 import os
+from pathlib import Path
+from typing import Optional, Union
 
 import numpy as np
 import torch
 import torch.nn as nn
 
-from .utils import ModelType
-
-
-class DisentNoiseModel(nn.Module):
-
-    def __init__(self, *nmodels):
-        """
-        Constructor.
-
-        This class receives as input a variable number of noise models, each one corresponding to a channel.
-        """
-        super().__init__()
-        # self.nmodels = nmodels
-        for i, nmodel in enumerate(nmodels):
-            if nmodel is not None:
-                self.add_module(f"nmodel_{i}", nmodel)
+from careamics.config import GMNMModel
 
-        self._nm_cnt = 0
-        for nmodel in nmodels:
-            if nmodel is not None:
-                self._nm_cnt += 1
+from .utils import ModelType
 
-        print(f"[{self.__class__.__name__}] Nmodels count:{self._nm_cnt}")
+# TODO this module shouldn't be in lvae folder
 
-    def likelihood(self, obs: torch.Tensor, signal: torch.Tensor) -> torch.Tensor:
 
-        if obs.shape[1] == 1:
-            assert signal.shape[1] == 1
-            assert self.n2model is None
-            return self.nmodel_0.likelihood(obs, signal)
+def noise_model_factory(
+    model_config: Union[GMNMModel, None], paths: Optional[list[Union[str, Path]]] = None
+) -> nn.Module:
+    """Noise model factory.
 
-        assert obs.shape[1] == self._nm_cnt, f"{obs.shape[1]} != {self._nm_cnt}"
-
-        ll_list = []
-        for ch_idx in range(obs.shape[1]):
-            nmodel = getattr(self, f"nmodel_{ch_idx}")
-            ll_list.append(
-                nmodel.likelihood(
-                    obs[:, ch_idx : ch_idx + 1], signal[:, ch_idx : ch_idx + 1]
-                )
+    Parameters
+    ----------
+    model_config : GaussianMixtureNoiseModel
+        _description_
+
+    Returns
+    -------
+    nn.Module
+        _description_
+
+    Raises
+    ------
+    NotImplementedError
+        _description_
+    """
+    if model_config and paths:
+        if model_config.model_type == "GaussianMixtureNoiseModel":
+            noise_models = []
+            for path in paths:
+                model_config.path = path
+                noise_models.append(GaussianMixtureNoiseModel(model_config))
```

Comment:
This is confusing. There is a chance that each instance of `GaussianMixtureNoiseModel` will have a reference to `model_config.path` so when it updates during the loop it will update in all the instances. Even if python does handle this by copying at assignment, it still does not seem like the best practice.

Reply:
Currently, they are not. The only change in the config is the path.

Reply:
But, certainly, one could handle it in a better way.